x-pack/plugins/ml/public/application/model_management/model_actions.tsx

```diff
@@ -33,6 +34,80 @@ import { isTestable, isDfaTrainedModel } from './test_models';
 import { ModelItem } from './models_list';
 import { usePermissionCheck } from '../capabilities/check_capabilities';
 
+const ViewTrainingDataAction: FC<{
+  item: ModelItem;
+}> = ({ item }) => {
+  const urlLocator = useMlLocator()!;
+  const {
+    services: {
+      application: { navigateToUrl },
+    },
+  } = useMlKibana();
+
+  const handleClick = async () => {
+    if (item.metadata?.analytics_config === undefined) return;
```

Comment:
Nit: Could this be a check probably with a type guard so we can avoid the `as` casting later on? (Saw these lines were copied as is but maybe it's an easy fix)